homeassistant/components/remember_the_milk/__init__.py

```diff
@@ -0,0 +1,253 @@
+            json.dump(self._config, config_file)
+
+    def get_token(self, profile_name):
+        """Get the server token for a profile."""
+        if profile_name in self._config:
+            return self._config[profile_name][CONF_TOKEN]
+        return None
+
+    def set_token(self, profile_name, token):
+        """Store a new server token for a profile."""
+        if profile_name not in self._config:
+            self._config[profile_name] = dict()
+        self._config[profile_name][CONF_TOKEN] = token
+        self.save_config()
+
+    def delete_token(self, profile_name):
+        """Delete a token for a profile.
+
+        Usually called when the token has expired.
+        """
+        self._config.pop(profile_name, None)
+        self.save_config()
+
+
+class RememberTheMilk(Entity):
+    """MVP implementation of an interface to Remember The Milk."""
+
+    def __init__(self, name, api_key, shared_secret, token, rtm_config):
+        """Create new instance of Remember The Milk component."""
+        import rtmapi
+
+        self._name = name
+        self._api_key = api_key
+        self._shared_secret = shared_secret
+        self._token = token
+        self._rtm_config = rtm_config
+        self._rtm_api = rtmapi.Rtm(api_key, shared_secret, "delete", token)
+        self._token_valid = None
+        self._check_token()
+        _LOGGER.debug("instance created for account %s", self._name)
+
+    def _check_token(self):
+        """Check if the API token is still valid.
+
+        If it is not valid any more, delete it from the configuration. This
+        will trigger a new authentication process.
+        """
+        valid = self._rtm_api.token_valid()
+        if not valid:
+            _LOGGER.error('Token for account %s is invalid. You need to '
+                          'register again!', self.name)
+            self._rtm_config.delete_token(self._name)
+            self._token_valid = False
+        else:
+            self._token_valid = True
+        return self._token_valid
+
+    def create_task(self, call):
+        """Create a new task on Remember The Milk.
+
+        You can use the smart syntax to define the attribues of a new task,
+        e.g. "my task #some_tag ^today" will add tag "some_tag" and set the
+        due date to today.
+        """
+        import rtmapi
+
+        try:
+            task_name = call.data.get('name')
+            result = self._rtm_api.rtm.timelines.create()
+            timeline = result.timeline.value
+            self._rtm_api.rtm.tasks.add(
+                timeline=timeline, name=task_name, parse='1')
+            _LOGGER.debug('created new task "%s" in account %s',
+                          task_name, self.name)
+        except rtmapi.RtmRequestFailedException as rtm_exception:
+            _LOGGER.error('Error creating new Remember The Milk task for '
+                          'account %s: %s', self._name,
+                          str(rtm_exception))
```

Comment:
no need to cast to string, %s will already take care of that.

Reply:
OK